### PR TITLE
fix(nuget): Parse nested Version XML elements

### LINKED ISSUE #7188: Nuget: support nested version tag inside packagereference
**What would you like Renovate to be able to do?**
Renovate currently supports PackageReferences in the format of:
`<PackageReference Include="Microsoft.ServiceFabric" Version="6.5.664" />`
but not in the format of:
`    <PackageReference Include="Microsoft.ServiceFabric">
      <Version>6.5.664</Version>
    </PackageReference>`

It would be nice if Renovate supported both formats.

### PR DESCRIPTION
**Pull request by zharinov**
Closes #7188
Example: https://github.com/zharinov/renovate-test-nuget/pull/2/files

### CODE REVIEW

File: lib/manager/nuget/extract.ts [hunk 1]
```diff
@@ -74,22 +74,73 @@ async function determineRegistryUrls(
...
+/**
+ * https://docs.microsoft.com/en-us/nuget/concepts/package-versioning
+ * This article mentions that  Nuget 3.x and later tries to restore the lowest possible version
+ * regarding to given version range.
+ * 1.3.4 equals [1.3.4,)
+ * Due to guarantee that an update of package version will result in its usage by the next restore + build operation,
+ * only following constrained versions make sense
+ * 1.3.4, [1.3.4], [1.3.4, ], [1.3.4, )
+ * The update of the right boundary does not make sense regarding to the lowest version restore rule,
+ * so we don't include it in the extracting regexp
```
Comment (rarkins) [inline lib/manager/nuget/extract.ts:86]: I'm ok with this change for joe but I think we'd be better to extract the full constraint and make our nuget versioning smarter instead

Comment (zharinov) [reply lib/manager/nuget/extract.ts:86]: Agree, let's do this in separate PR

Comment (zharinov) [reply lib/manager/nuget/extract.ts:86]: Created an issue for this: https://github.com/renovatebot/renovate/issues/7191

**Review by viceice — COMMENTED** [review]
I think we can improve this, because we are only need to check a specific xml level and not all elements.

eg: `Project > ItemGroup > PackageReference`. so we can break early.

**Review by viceice — APPROVED** [review]

**Pull request merged**